pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1787,16 +1787,17 @@ public void updateUserConfig(UserConfig userConfig)
   public void updateTableConfig(TableConfig tableConfig)
       throws IOException {
     validateTableTenantConfig(tableConfig);
-    setExistingTableConfig(tableConfig);
+    setExistingTableConfig(tableConfig, -1);
   }
 
   /**
-   * Sets the given table config into zookeeper
+   * Sets the given table config into zookeeper with the expected version, which is the previous tableConfig znRecord
+   * version. If the expected version is -1, the version check is ignored.
    */
-  public void setExistingTableConfig(TableConfig tableConfig)
+  public void setExistingTableConfig(TableConfig tableConfig, int expectedVersion)
```

Comment:
Let's keep the existing one and add one with `expectedVersion`, no need to change the existing usages